Check for force before check for installed (Fix #256)

alikins opened the pull request:
##### SUMMARY
Check for force before check for installed

The 'and not force_overwrite' would remove everything from
the requirements_to_install list if --force was used.
Move force_overwite checking condition before the 'or'
already installed check.

Fixes: #256

WIP=needs unit tests



##### ISSUE TYPE

 - Bugfix Pull Request



##### MAZER VERSION
<!--- Paste verbatim output from "mazer version" between quotes below -->
```
name = mazer
version = 0.5.0
config_file = /home/adrian/.ansible/mazer.yml
uname = Linux, newswoop, 5.0.5-200.fc29.x86_64, #1 SMP Wed Mar 27 20:58:04 UTC 2019, x86_64
executable_location = /home/adrian/venvs/mazer_0.4.0_py36/bin/mazer
python_version = 3.6.8 (default, Jan 27 2019, 09:00:23) [GCC 8.2.1 20181215 (Red Hat 8.2.1-6)]
python_executable = /home/adrian/venvs/mazer_0.4.0_py36/bin/python


```


##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful. -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

alikins committed:
Check for force before check for installed

The 'and not force_overwrite' would remove everything from
the requirements_to_install list if --force was used.
Move force_overwite checking condition before the 'or'
already installed check.

Fixes: #256

ironfroggy commented on ansible_galaxy/actions/install.py:
```diff
@@ -79,8 +79,7 @@ def install_repositories_matching_repository_specs(galaxy_context,
     # This filters out already installed repositories unless --force. Aside from the warning, 'mazer install alikins.something_installed_already' is ok.
-    requirements_to_install = [y for y in requirements_list if y.requirement_spec not in already_installed_repository_spec_set and not force_overwrite]
-
+    requirements_to_install = [y for y in requirements_list if force_overwrite or y.requirement_spec not in already_installed_repository_spec_set]
```
If you have a complex/multi-part conditional maybe a list comprehension isn't the most readable format to put that logic in, if we want to avoid future mishaps.

alikins replied:
Yeah, had the same thought. May just if/else on the 'force' before the comprehension.

[**edit**: rm'ed comment about the wrong bug]

alikins replied:
At some point, when the install bits is more 'transaction' oriented, it's likely the 'force' option will need to be set per collection (instead of overall), but that isn't the case now, so no reason not to mv the 'if force' outside.

alikins committed:
Move galaxy_context fixture up a level

Move galaxy_context up to tests/conftest.py
so ansible_galaxy_cli tests can use it.

alikins committed:
Fix test_execute_remove that --force fix broke

test_execute_remove() was using cli 'install --force'
as setup for testing remove, but since '--force'
was previously broken, nothing was ever installed,
so test appeared to work since the removed content
didn't exist after the test. But the test was never
testing remove much.
